src/Database/SqlDialectTrait.php

```diff
@@ -206,6 +206,14 @@ protected function _updateQueryTranslator($query)
      */
     protected function _removeAliasesFromConditions($query)
     {
+        if (!empty($query->clause('join'))) {
+            trigger_error(
```

Comment:
i can haz exception or nothing - E_USER_NOTICE would just be annoying from monitoring perspective in newrelic during production :)